cfgov/cfgov/settings/production.py

```diff
@@ -95,3 +99,75 @@
         "Environment variable ALLOWED_HOSTS is either not defined or is "
         "not valid JSON. Expected a JSON array of allowed hostnames."
     )
+
+# SAML2 Authentication
+#
+# Requires the SAML_AUTH, SAML_ROOT_URL, SAML_ENTITY_ID, and SAML_METADATA_URL
+# environment variables to be defined.
+#
+# - SAML_AUTH: Enable SAML2 authentication with a value of "True".
+# - SAML_ROOT_URL: The URL that browsers will use to access this site. The
+#   SAML2 relay URL will be constructed from this, and it must match what is
+#   configured for this site in the SAML2 identity provider.
+# - SAML_ENTITY_ID: The entity id configured for this service provider in the
+#   SAML2 identity provider.
+# - SAML_METADATA_URL: The remote URL of the identity provider's metadata for
+#   this service provider.
+SAML_AUTH = os.environ.get("SAML_AUTH") == "True"
+if SAML_AUTH:
+    # Update built-in Django settings for SAML authetnication
+    INSTALLED_APPS += ('djangosaml2',)
+    MIDDLEWARE += ('djangosaml2.middleware.SamlSessionMiddleware',)
+    AUTHENTICATION_BACKENDS = (
+        'django.contrib.auth.backends.ModelBackend',
+        'djangosaml2.backends.Saml2Backend',
+    )
+    LOGIN_URL = '/saml2/login/'
+    SESSION_EXPIRE_AT_BROWSER_CLOSE = True
+
+    # Map Django attributes to our SAML identity provider
+    SAML_DJANGO_USER_MAIN_ATTRIBUTE = 'email'
+    SAML_DJANGO_USER_MAIN_ATTRIBUTE_LOOKUP = '__iexact'
+    SAML_ATTRIBUTE_MAPPING = {
+        'emailAddress': ('email', ),
+    }
+    SAML_CREATE_UNKNOWN_USER = False
+
+    # URL lookups
+    SAML_ROOT_URL = os.environ.get("SAML_ROOT_URL")
```

Comment:
Is it worth changing these various `os.environ.get()` to `os.environ[]`, to deliberately raise an exception if the SAML specific environment variables aren't set, if `SAML_AUTH` is? Otherwise, these settings will be set to None, which will presumably raise some unpredictable error downstream. By using `[]`, we'd know right away because the app won't even start up.

Reply:
Done!